src/components/VideoThumbnail/VideoThumbnail.tsx

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+
+import {useI18n} from '../../utilities/i18n';
+import {
+  secondsToTimeComponents,
+  secondsToTimestamp,
+  secondsToDurationTranslationKey,
+} from '../../utilities/duration';
+
+import {PlayIcon} from './illustrations';
+import styles from './VideoThumbnail.scss';
+
+export interface VideoThumbnailProps {
+  /** URL source for thumbnail image. */
+  thumbnailUrl: string;
+  /** Length of video in seconds. */
+  videoLength?: number;
+  /** Custom ARIA label for play button.
+   * @default 'Play video of length {human readable duration}'
+   */
+  accessibilityLabel?: string;
+  /** Callback on click or keypress of thumbnail. Use to trigger render of the video player in your chosen format, for example within a modal or fullscreen container. */
+  onClick(): void;
+  /** Callback on mouse enter, focus, or touch start of thumbnail. Use to trigger video preload. */
+  onBeforeStartPlaying?(): void;
+}
+
+export const VideoThumbnail = ({
+  thumbnailUrl,
+  videoLength,
+  accessibilityLabel,
+  onClick,
+  onBeforeStartPlaying,
+}: VideoThumbnailProps) => {
+  const i18n = useI18n();
+  let defaultLabel = i18n.translate(
+    'Polaris.VideoThumbnail.playButtonA11yLabel.default',
+  );
+
+  if (videoLength) {
+    const {hours, minutes, seconds} = secondsToTimeComponents(videoLength);
+
+    defaultLabel = i18n.translate(
+      'Polaris.VideoThumbnail.playButtonA11yLabel.defaultWithDuration',
+      {
+        duration: i18n.translate(secondsToDurationTranslationKey(videoLength), {
+          hourCount: hours,
+          minuteCount: minutes,
+          secondCount: seconds,
+        }),
+      },
+    );
+  }
+
+  const buttonLabel = accessibilityLabel ? accessibilityLabel : defaultLabel;
```

Comment:
A little petty but this means that we can make 2 or 3 calls to i18n depending on if there is accessibilityLabel and videoLength are set. Is there a way where we always only make one call to i18n in all scenarios (it's not thaaaat expensive, but why do work we don't need to)?

```
let buttonLabel;
if (accessibilityLabel) {
  buttonLabel = accessibilityLabel;
} else if (videoLength) {
  const {hours, minutes, seconds} = secondsToTimeComponents(videoLength);

  buttonLabel = i18n.translate(
      'Polaris.VideoThumbnail.playButtonA11yLabel.defaultWithDuration',
      {
        duration: i18n.translate(secondsToDurationTranslationKey(videoLength), {
          hourCount: hours,
          minuteCount: minutes,
          secondCount: seconds,
        }),
      },
  );
} else {
  buttonLabel = i18n.translate(
    'Polaris.VideoThumbnail.playButtonA11yLabel.default',
  );
}
```

kinda thing